### DISCUSSION TITLE
Disable TRV command when another Vtherm is not heating

### DISCUSSION THREAD

**Discussion by N0L12** [jmcollin78/versatile_thermostat#744] [Q&A]
Hey everyone, I happily :) setup my radiator heating with 2 Vtherm: one "over switch" is to start/stop the radiator circuit of the gas boiler controlling the temperature of the living rooms, the second "over valve" is only dedicated to control the TRVs installed on the bedroom radiators.
Because it is useless to continuously act on the TRVs (wasting battery life) if the boiler is not pumping hot water, is there a setting to enable the command of the second VT only when the first VT is in "heating" mode?

**Comment by jmcollin78** [comment]
Hello,

You should have a look on this: https://github.com/jmcollin78/versatile_thermostat/blob/main/documentation/en/feature-central-boiler.md

**Reply by N0L12** [reply]
Thank you, for the prompt answer. I've deeply considered the "central boiler" feature, but it appears to be useful to command the boiler when the valves need it, but I cannot find a way to make it work the other way around.

At the moment here is what I did to disable the TRV command when the boiler is off:
-created a "number" entity
-set this new entity as the device to be controlled from the second VTherm "over valve" (instead of the TRV entities)
-created an automation to copy the value from this new entity to the TRV "opening %" entities ONLY when the main Vtherm is in "heating" action.

With this turnaround it is working as expected.

**Reply by stedaho** [reply]
I have the same request (for a different use case):
I am controlling the TRVs of a part-time home with your lovely add-on. Now, in the winter's season, I want to set all TRVs (Sonoff TRVZB) to a fixed opening/closing percentage and just turn on the central boiler depending on minimum indoor and outdoor temperature to a) improve battery life (no motor usage during long periods of absence) and b) save energy by turning off the water pump as long as possible.
I tried turning off the VTherms and set the opening/closing percentages of the underlying TRVs directly, but turning off the VTHerms also turns off the underlying TRVs and apparently they do not accept the opening/closing percentages then. 
So the workaround of @N0L12 could be a possible solution for me, but as I am using many TRVs and would have to do it for both opening and closing percentage this would be a lot of effort for such a hack.
Another solution for me (and maybe also for you, @N0L12 ?) could be to fully disable the VTherms, but I don't know if it is a good idea...

**Reply by jmcollin78** [reply]
And why not setting the setpoing to let's say Anti frozen or Eco ?

**Reply by stedaho** [reply]
> And why not setting the setpoing to let's say Anti frozen or Eco ?

First, let me thank you for all the time and effort you are putting in the development of this very helpful plugin!

Second, I am currently running the VTherms in Frost mode and it technically works, so no real issues here.
But there are three concerns why I'd like to change it during my long times of absence.
When I am not there, there is no comfort and therefore no exact regulation needed. That's why I would prefer to set the TVRs to a fixed opening percentage (to save battery power) und just want to turn on the central boiler for a short period of time just once or twice a day (to save energy due to running the water pump the whole day while the boiler is running at 10% power).

But the main reason why I want to implement it like that is the following:
In December I had some hardware issues with my Home Assistent installation. This resulted in a total failure of my Zigbee connection so VTherm/HASS could not control the TVRs. If this would have happened during the frost period (and while the TVRs are closed), my safety logic of the central boiler would turn it on, but cannot heat the building to the "disabled" radiators, which obviously would cause severe problems.
That's why I thought about a fixed valve position during my absence.

**Reply by jmcollin78** [reply]
What are your TRV ? Can you control the valve directly ?

If yes I suggest something very simple when you want to activate this "long sleep mode":
1. turn-off the VTherm (it will not turn-on itself by construction) so that Vtherm will no more act on the valve
2. set the valve openness to the value you want (maybe the value of an input_number). This can be done directly by calling the number/set_value action (try it into Dev Tools / Action).

Put that in an automation and start this automation when you leave for a long time.

This is not so easy to do with VTherm:
1. VTherm only deals with temperatures,
2. Configuration UI is not adapted for that, so I will need to add a new number entity,
3. setting the valve open degree is done by self-regulation algorithm, I should bybass to do what you think about.

and after all, this is a very special case with have nothing to do with temperature regulation (the core feature of Vtherm).

I hope you understand and what I says is clear (my english is not my best skill).

**Reply by stedaho** [reply]
Thanks for your quick response!

I use the Sonoff TRVZB (and so was very happy when you added the direct control end of last year, by the way!)

When trying your suggested procedure, I unfortunately ran into the problem that the TRVs do not open the valve when they are turned off. So I have to at least turn the underlying climate entity on - which automatically activates the VTherm again.

I understand that this is a very special use case (which is why I did not open a "real" feature request for it), but when I stumbled on this discussion I thought it could be helpful to add another use case for the original request.

By the way, I added the workaround @N0L12 suggested (pass valve opening through a number entity) for one TVR this evening and it seems to work fine, so this would be the way to go for me if you do not have another idea!

**Comment by jmcollin78** [comment]
> When trying your suggested procedure, I unfortunately ran into the problem that the TRVs do not open the valve when they are turned off. So I have to at least turn the underlying climate entity on - which automatically activates the VTherm again.

ok. Yes, you should deactivate the "follow underlyin..." explained here before doing that: https://github.com/jmcollin78/versatile_thermostat/blob/main/documentation/en/over-climate.md#follow-underlying-temperature-changes

If the checkbox of the "follow..." entity is off, then you will be able to start the underlying without starting the VTherm.

**Reply by stedaho** [reply]
I am pretty sure that I tried it and it did not make a difference - but I will double check it tomorrow and give feedback!

**Reply by stedaho** [reply]
Here are the results of my test as promised. Left is VTherm, right is the default climate control created by ZHA.

Starting point:
![image](https://github.com/user-attachments/assets/f5a9a959-4264-4653-991a-9c439b344606)

Disabling "Follow underlying...":
![image](https://github.com/user-attachments/assets/8648d129-1e0f-40a5-9a1a-56aebe054b88)

Turn of VTherm results in turning off the default climate control:
![image](https://github.com/user-attachments/assets/b28fbca9-056f-4e3c-b282-52dd6e81b08e)

Turning on the underlying climate again to manually force valve position:
![image](https://github.com/user-attachments/assets/332412b3-c29b-4175-a3c1-2aee41b1b275)

As you see, the VTherm ist turned on again (although the "Follow underlying temp change" option is off). So this trick does not help.


I had another idea last night. If VTherm would support a _"Maximum opening degree"_ (according to the newly introduced "Minimum opening degree") and would allow to change that programmatically, I could set both minimum and maximum to the same value and could force a specific position. What do you think about that?

But for now, I am fine with @N0L12 's solution (proxy number)!

**Comment by jmcollin78** [comment]
Hello,

It must be related to my eyes, but the Vtherm is NOT turned on and stays always off:

<img width="79" alt="Capture d’écran 2025-01-09 à 22 32 27" src="https://github.com/user-attachments/assets/14d799b5-2beb-420f-a4d2-fb4369e3e23c" />

Off means it is Off. Heat means the underlying climate is heating because you turn it on manually. What did you expect ?

VTherm cannot prevent you to turn on your device manually. The "follow ..." checkbox, just tells Vtherm to follow to underlyings changes but cannot prevent you to change it manually.

**Reply by stedaho** [reply]
I think you are not right.
Please compare my second and third screenshot.

In the second screenshot you see "Off (Heat) - Frost". This means, thst VTherm would start heating when the temperature drops below the setpoint. So VTherm is not really off, but currently not heating.
In my third screenshot you see "Off - Frost". In this mode, VTherm is really off and would not interfere with my manual opening degree settings (which is what I want to achieve).

But as I wrote, the solution with the number proxy works fine!

**Comment by jmcollin78** [comment]
Hello @stedaho ,

Sorry but: 

> In the second screenshot you see "Off (Heat) - Frost". This means, thst VTherm would start heating when the temperature drops below the setpoint. So VTherm is not really off, but currently not heating.

This is a wrong asumption. if VTherm is off it is off and will not "start heating when the temperature drops". A Vtherm NEVER starts alone without a user action. (The only exception is the auto-start/stop feature describe here: https://github.com/jmcollin78/versatile_thermostat/blob/main/documentation/en/feature-auto-start-stop.md)
I can't understand why you are so convinced of this. When VTherm is off, you can do what you want with the underlying heater without interfering with the Vtherm state. Only have_action ('heating') can be set because this is mapped to the hvac_action state of the underlying if it is available.

> In my third screenshot you see "Off - Frost". In this mode, VTherm is really off and would not interfere with my manual opening degree settings (which is what I want to achieve).

This asumption is right. Vtherm is off and will not interfere, like the previous one.

**Reply by stedaho** [reply]
Thanks for your reply. As you wrote the code, you know better than me what is happening! :-)
May you explain me the difference between the two modes "Off (Heat) - Frost" and "Off - Frost"? And what does the "Heat" mean if it is not heating?

**Reply by jmcollin78** [reply]
In a Climate entity in HA there at least 3 informations, hvac_mode and hvac_action and preset:
1. hvac_mode tells in which mode is the climate. This can be 'Heat', 'Cold', 'Off', 'Fan', ...
2. hvac_action tells if the device is 'heating' or 'cooling' or 'idle' or ...
3. preset tells the preset temperature currently used as setpoint (if any)

In the 'Off (Heat) - Frost' string (which is a HA string and not a Vtherm one) this means : hvac_mode is off, hvac_action is heating and preset is Frost.
I suppose 'Off - Frost' means hvac_mode is Off, hvac_action is idle (no action) and preset is Frost.

With VTherm `over_climate`, the VTherm can be off but the underlying climate device is 'on' and heating. That is possible if you start manually the underlying device (which is your case if I remember well).
